### PR TITLE
27 machinesubscriber

### LINKED ISSUE #27: MachineSubscriber
Implement the SubscribeService interface.
The component should use the MachineDataRepository and the MachineData entity class.

### PR DESCRIPTION
**Pull request by Lawst96**
Closes #27 

## Description
Involves implementation of the subscription module. It's event-driven and only reacts when a new value is written on the machine and pushed to the subscription thread. Multiple subscriptions can be handled at the same time.
Some nodes had to be changed in their enum definitions, as they were incorrect. 
Copy constructors added to some entities, to allow for basing new objects on an existing object, without copying the reference.

There is no test for the Subscription class. This is because it relies heavily on external systems and would require extensive mocking. It's, by all means, possible, but deemed too costly to implement at this point in time.

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes, if necessary.
- [x] All tests pass (existing and potential new).
- [x] Sonar check has passed
- [x] No codesmells or bugs

## Other Relevant Information
Provide any other important details below.

### CODE REVIEW

File: BrewMesModulesParent/common_repository/src/main/java/com/brewmes/common_repository/BatchRepository.java [hunk 1]
```diff
@@ -4,6 +4,9 @@
 import org.springframework.data.mongodb.repository.MongoRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface BatchRepository extends MongoRepository<Batch, String> {
+    List<Batch> findByConnectionID(String connectionID);
```
Comment (Kaszz) [inline BrewMesModulesParent/common_repository/src/main/java/com/brewmes/common_repository/BatchRepository.java:11]: Det ligner ikke den bliver brugt længere?

File: BrewMesModulesParent/subscriber/pom.xml [hunk 2]
```diff
@@ -17,6 +17,23 @@
             <artifactId>stack-core</artifactId>
             <version>0.5.2</version>
         </dependency>
+        <dependency>
+            <groupId>com.brewmes</groupId>
+            <artifactId>common</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (Kaszz) [inline BrewMesModulesParent/subscriber/pom.xml:23]:
```suggestion
            <version>${project.version}</version>
```

File: BrewMesModulesParent/subscriber/pom.xml [hunk 3]
```diff
@@ -17,6 +17,23 @@
...
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.brewmes</groupId>
+            <artifactId>common_repository</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (Kaszz) [inline BrewMesModulesParent/subscriber/pom.xml:29]:
```suggestion
            <version>${project.version}</version>
```

File: BrewMesModulesParent/api/pom.xml [hunk 4]
```diff
@@ -26,6 +26,12 @@
 			<artifactId>gson</artifactId>
 			<version>2.8.6</version>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>com.brewmes</groupId>
+            <artifactId>common_repository</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (Kaszz) [inline BrewMesModulesParent/api/pom.xml:32]:
```suggestion
            <version>${project.version}</version>
```

**Review by Wylardt — APPROVED** [review]
Good job guys

**Review by Kechr — CHANGES_REQUESTED** [review]
Haven't reviewed tests yet, as I saw you made some changes that haven't been pushed yet.

File: BrewMesModulesParent/subscriber/src/test/java/com/brewmes/subscriber/MachineSubscriberTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        MachineData data = subscribeService.getLatestMachineData(FAKE_ID);
+        assertNull(data);
+    }
+
+    @Test
+    void getLatestMachineDataSuccess() {
+        MachineData fakeData = new MachineData();
+        fakeData.setNormSpeed(123.0);
+        SubscriptionStub subscriptionStub = new SubscriptionStub();
+        subscriptionStub.latestMachineData = fakeData;
```
Comment (Kechr) [inline BrewMesModulesParent/subscriber/src/test/java/com/brewmes/subscriber/MachineSubscriberTest.java:108]: Are these used in this test?

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/util/MachineState.java [hunk 6]
```diff
@@ -30,4 +30,13 @@
...
     }
+
+    public static MachineState getStateFromValue(int value) {
+        for (MachineState machineState : MachineState.values()) {
+            if (machineState.value == value){
+                return machineState;
+            }
+        }
+        return null;
+    }
```
Comment (Kechr) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/util/MachineState.java:41]: This method needs to be tested.

**Review by Kaszz — APPROVED** [review]
Hvis den metode Kenneth nævner bliver testet er det👍🏼

**Review by Teyson — APPROVED** [review]

**Review by Kechr — APPROVED** [review]

**Pull request merged**